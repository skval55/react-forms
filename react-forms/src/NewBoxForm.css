.BoxForm {
    margin:auto;
   display: flex;
   flex-direction: column;
   width: 50%;
   
}

.BoxForm-label{
    font-size: large;
    font-weight: bold;
}

.BoxForm-input{
    text-align: center;
    border: none;
}

.BoxForm-input:focus{
    outline: none;
    background-color: rgb(191, 230, 242);
}

.BoxForm-button{
    border-radius: 8px;
    border: none;
    margin: auto;
    margin-top: 10px;
    width: 25%;
}

.BoxForm-button:hover{
    background-color: rgb(170, 170, 170);
}
