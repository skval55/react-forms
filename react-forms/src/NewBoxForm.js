import React, { useState } from "react";
import "./NewBoxForm.css";

const NewBoxForm = ({ addBox }) => {
  const INITIAL_STATE = {
    height: "",
    width: "",
    backgroundColor: "",
  };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleSubmit = (e) => {
    e.preventDefault();
    addBox(formData);
    setFormData(INITIAL_STATE);
  };

  const handleChange = (e) => {
    let { name, value } = e.target;
    if (name === "height" || name === "width") {
      value = Number(value);
      if (isNaN(value)) {
        value = "";
      }
    }
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  return (
    <form className="BoxForm" onSubmit={handleSubmit}>
      <label className="BoxForm-label" htmlFor="height">
        height:
      </label>
      <input
        className="BoxForm-input"
        id="height"
        name="height"
        value={formData.height}
        onChange={handleChange}
      />

      <label className="BoxForm-label" htmlFor="width">
        width:
      </label>
      <input
        className="BoxForm-input"
        id="width"
        name="width"
        value={formData.width}
        onChange={handleChange}
      />

      <label className="BoxForm-label" htmlFor="backgroundColor">
        Background Color:
      </label>
      <input
        className="BoxForm-input"
        id="backgroundColor"
        name="backgroundColor"
        value={formData.backgroundColor}
        onChange={handleChange}
      />
      <button className="BoxForm-button">Add a new box</button>
    </form>
  );
};

export default NewBoxForm;
