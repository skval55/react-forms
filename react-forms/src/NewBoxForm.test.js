import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

// smoke test
test("renders without crashing", () => {
  render(<NewBoxForm addBox={() => console.log("test")} />);
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <NewBoxForm addBox={() => console.log("test")} />
  );
  expect(asFragment()).toMatchSnapshot();
});
