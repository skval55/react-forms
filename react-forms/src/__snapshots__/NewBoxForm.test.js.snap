// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="BoxForm"
  >
    <label
      class="BoxForm-label"
      for="height"
    >
      height:
    </label>
    <input
      class="BoxForm-input"
      id="height"
      name="height"
      value=""
    />
    <label
      class="BoxForm-label"
      for="width"
    >
      width:
    </label>
    <input
      class="BoxForm-input"
      id="width"
      name="width"
      value=""
    />
    <label
      class="BoxForm-label"
      for="backgroundColor"
    >
      Background Color:
    </label>
    <input
      class="BoxForm-input"
      id="backgroundColor"
      name="backgroundColor"
      value=""
    />
    <button
      class="BoxForm-button"
    >
      Add a new box
    </button>
  </form>
</DocumentFragment>
`;
